src/Powercord/plugins/pc-emojiUtility/index.js

```diff
@@ -50,20 +181,218 @@ module.exports = class EmojiUtility extends Plugin {
         '{c} [emote name]',
         (args) => {
           const argument = args.join(' ').toLowerCase();
+          if (argument.length === 0) {
+            return {
+              send: false,
+              result: this.settings.get('useEmbeds')
+                ? {
+                  type: 'rich',
+                  description: 'Please provide an emote name',
+                  color: 16711680
+                }
+                : 'Please provide an emote name'
+            };
+          }
+
           const emojis = Object.values(emojiStore.getGuilds()).flatMap(r => r.emojis);
 
           const foundEmojis = emojis.filter(emoji => emoji.name.toLowerCase().includes(argument));
           if (foundEmojis.length > 0) {
+            const emojisAsString = foundEmojis.map(emoji => this.getFullEmoji(emoji)).join('');
+            if (emojisAsString.length > 2000) {
+              return {
+                send: false,
+                result: `That is more than 2000 characters, let me send that locally instead!\n${emojisAsString}`
+              };
+            }
+
             return {
               send: true,
-              result: foundEmojis.map(emoji => `<${(emoji.animated ? 'a' : '') + emoji.allNamesString + emoji.id}>`).join('')
+              result: emojisAsString
             };
           }
+
           return {
             send: false,
-            result: `Could not find any emotes containing **${argument}**`
+            result: this.settings.get('useEmbeds')
+              ? {
+                type: 'rich',
+                description: `Could not find any emotes containing **${argument}**`,
+                color: 16711680
+              }
+              : `Could not find any emotes containing **${argument}**`
           };
         }
       );
+
+    powercord
+      .pluginManager
+      .get('pc-commands')
+      .register(
+        'saveemote',
```

Comment:
I'm not really sure if this is really something needed. Discord already allows you to open an emote in browser and download it. Imo, it would be util if this was able to re-upload emojis automatically to a server instead of letting ppl re-upload them

Reply:
Going to make it context later for now it can be a command

![image](https://user-images.githubusercontent.com/11248267/51127077-f03e1300-1824-11e9-82b3-f1e2eb9635a4.png)